tf/tfprocess.py

```diff
@@ -172,14 +177,28 @@ def __init__(self, cfg):
                                        trainable=False,
                                        dtype=tf.int64)
 
-    def init_v2(self, train_dataset, test_dataset, validation_dataset=None):
-        self.train_dataset = train_dataset
-        self.train_iter = iter(train_dataset)
-        self.test_dataset = test_dataset
-        self.test_iter = iter(test_dataset)
-        self.validation_dataset = validation_dataset
-        self.init_net_v2()
-
+    def init_v2(self, train_dataset, test_dataset, validation_dataset=None):    
+        if self.strategy is not None:
+            self.train_dataset = self.strategy.experimental_distribute_dataset(train_dataset)
+        else:
+            self.train_dataset = train_dataset
+        self.train_iter = iter(self.train_dataset)
+        if self.strategy is not None:
+            self.test_dataset = self.strategy.experimental_distribute_dataset(test_dataset)
+        else:
+            self.test_dataset = train_dataset
+        self.test_iter = iter(self.test_dataset)
+        if self.strategy is not None and validation_dataset is not None:
+            self.validation_dataset = self.strategy.experimental_distribute_dataset(validation_dataset)
+        else:
+            self.validation_dataset = validation_dataset
+        this = self
```

Comment:
Curious trick - I think you can move it inside the if block though?

Reply:
some java thing I picked up some time ago :p Anyways moved.